pinot-query-planner/src/main/java/org/apache/pinot/query/planner/serde/PlanNodeDeserializer.java

```diff
@@ -117,8 +117,18 @@ private static MailboxReceiveNode deserializeMailboxReceiveNode(Plan.PlanNode pr
 
   private static MailboxSendNode deserializeMailboxSendNode(Plan.PlanNode protoNode) {
     Plan.MailboxSendNode protoMailboxSendNode = protoNode.getMailboxSendNode();
+
+    List<Integer> receiverIds;
+    List<Integer> protoReceiverIds = protoMailboxSendNode.getReceiverStageIdsList();
+    if (protoReceiverIds == null || protoReceiverIds.isEmpty()) {
```

Comment:
The `null` case should never be possible right? Even in the case of an older broker that is serializing using the older proto message without this field, the default value will be populated here on deserialization which should be empty list for `repeated int32`.

Reply:
nullability is tricky in protobuf. Depending on different configurations, null may or may not be present. That is why I prefer to be cautious and check for nulls just in case something change in the future.